### PR TITLE
Issue #1130: expose gated workspace diagnostics

### LINKED ISSUE #1130: [Agent] Gate Business/TPP Policy Surfaces By Trip Mode And Translate Policy State Into User Language
Target repo: `trip-planner`

## Why

The TPP integration is important for business travel, but leisure travelers should not see approval packet, policy posture, or proposal lifecycle language unless it is relevant. Business users need plain-language policy readiness, while developers still need the raw TPP state for debugging.

## Scope

Update workspace view-model logic and UI rendering so business/TPP surfaces are mode-aware and user-facing. Move raw TPP transport, policy IDs, proposal state IDs, and failure payloads into debug/advanced details.

## Non-Goals

- Changing TPP policy calculations.
- Removing proposal submission, polling, or result ingestion.
- Hiding business approval readiness from business trips.
- Updating the TPP portal UI itself.

## Tasks

- [ ] Define mode-aware rules for when policy, budget, proposal, and approval-readiness panels appear.
- [ ] Translate policy/proposal states into traveler or business-user labels such as "ready for approval", "needs exception", or "waiting for policy review".
- [ ] Remove policy posture from leisure route-comparison metrics unless a leisure trip explicitly has active policy state.
- [ ] Move raw TPP identifiers, transport statuses, polling details, and result payloads behind debug/advanced affordances.
- [ ] Add backend tests for leisure, business, and mixed edge cases.
- [ ] Add frontend tests that leisure workspaces do not render approval widgets and business workspaces still render readiness state.

## Acceptance Criteria

- [ ] Leisure workspaces without active policy state do not show policy posture, proposal lifecycle, approval packet, or TPP labels in normal UI.
- [ ] Business workspaces show approval readiness and required follow-up in plain language.
- [ ] Raw policy/proposal diagnostics remain available for debug inspection.
- [ ] Route comparison metrics do not include policy posture for normal leisure trips.
- [ ] Tests cover the visibility rules for leisure and business trips.

## Implementation Notes

Likely files:

- `trip_planner/app/services/workspace.py`
- `trip_planner/app/services/policy.py`
- `trip_planner/app/services/proposal.py`
- `trip_planner/app/schemas/workspace.py`
- `frontend/src/routes/WorkspacePage.tsx`
- `frontend/src/components/workspace/ScenarioComparison.tsx`
- `tests/app/test_workspace.py`
- `frontend/src/routes/WorkspacePage.test.tsx`

This issue should build on Issue 1's product/debug split rather than adding more one-off conditional rendering.

### PR DESCRIPTION
**Pull request by stranske**
### Keepalive: OFF

Closes #1130

## Summary
- lazy-render advanced workspace diagnostics only after the user opens the disclosure, keeping normal traveler/business copy free of raw policy IDs
- add a business-mode workspace test proving approval readiness remains visible in user-facing language
- verify raw proposal diagnostics become available through the advanced diagnostics affordance

## Validation
- `npm --prefix frontend test -- src/routes/WorkspacePage.test.tsx`
- `npm --prefix frontend run build`
- `pytest -q tests/app/test_workspace.py::test_workspace_endpoint_includes_typed_view_model_for_leisure_trip tests/app/test_workspace.py::test_workspace_endpoint_includes_typed_view_model_for_business_trip tests/app/test_workspace.py::test_workspace_view_model_keeps_active_leisure_policy_state_visible tests/app/test_workspace.py::test_workspace_view_model_debug_sections_preserve_raw_payload_shapes`
- `git diff --check`

### CODE REVIEW

File: frontend/src/routes/WorkspacePage.tsx [hunk 1]
```diff
@@ -800,6 +801,7 @@ function WorkspacePageContent({
   useEffect(() => {
     setCurrentWorkspace(workspace);
     setSelectedScenarioId(resolveMapScenarioId(workspace));
+    setShowWorkspaceDebugDetails(false);
```
[the remark this answers is not in the document]
Comment (stranske) [reply frontend/src/routes/WorkspacePage.tsx:804]: Addressed in 50ffb3c by controlling the disclosure with open={showWorkspaceDebugDetails}, so workspace changes close the element and keep the visible disclosure state aligned with rendered debug content.

**Comment by stranske** [comment]
Updated #1156 to address the Runtime CI failure.\n\nRoot cause: WorkspacePage tests still expected raw proposal/policy diagnostic payload details to be visible as normal UI after this branch moved those details behind Advanced diagnostics.\n\nChange: Updated the assertions so user-facing approval/follow-up guidance remains visible, raw diagnostic payload details are hidden by default, and those raw details appear after opening Advanced diagnostics.\n\nLocal validation:\n- `npm --prefix frontend test -- src/routes/WorkspacePage.test.tsx`\n- `npm --prefix frontend test`\n- `git diff --check`

**Comment by stranske** [comment]
Closer review-thread disposition: the Copilot disclosure-state thread is resolved on current head `5329f31b`. `WorkspacePage.tsx` controls the advanced diagnostics disclosure with `open={showWorkspaceDebugDetails}` and resets that state on workspace changes, so the visible disclosure state and rendered debug sections stay aligned.

Validation run this closer round:
- `git -C trip-planner-issue-1130-policy diff --check` -> passed
- `npm --prefix trip-planner-issue-1130-policy/frontend test -- src/routes/WorkspacePage.test.tsx` -> 37 tests passed

Not merging yet: the combined commit status for `Gate / gate` is still `pending` even though the individual Gate check runs on the current head show success; leaving merge for a later round once the required status context is no longer pending.

**Comment by stranske** [comment]
Closer update: addressed the Workspace diagnostics disclosure review thread in c9bddf0ea.

Evidence:
- The diagnostics `<details>` now has a workspace/proposal/debug-section remount key in addition to the controlled `open` state, so workspace changes close the browser disclosure and hidden payload together.
- Added regression coverage that opens Advanced diagnostics, rerenders with a changed workspace payload, and confirms the disclosure is closed and the next proposal payload is not exposed.
- Validation: `npm --prefix frontend test -- WorkspacePage.test.tsx` passed (38 tests); `npm --prefix frontend run build` passed; `git diff --check` passed.

Post-push state: GraphQL reports 0 unresolved review threads on PR #1156. Checks are queued/pending on the new head c9bddf0ea, so the next closer pass should merge only after CI settles green.
